### PR TITLE
Add functions to get specific file formats

### PR DESCRIPTION
**Pull request by nickpeihl**
The EMS file service supports multiple geospatial formats for file services. This PR introduces several new functions for retrieving a specific format for a layer. If the requested format is unavailable it falls back to the default format.

This PR is required for https://github.com/elastic/ems-landing-page/issues/255.

See also https://github.com/elastic/ems-file-service/issues/191

### CODE REVIEW

**Comment by jsanz** [comment]
@nickpeihl should we work on finishing this PR before releasing `v7.11`?

File: src/file_layer.ts [hunk 1]
```diff
@@ -110,4 +141,14 @@ export class FileLayer extends AbstractEmsService {
...
   }
+
+  private _getFormatOfType(type: EMSFormatTypeStrings): EMSFormats {
+    const requestedFormat = this._config.formats.find((format) => {
+      return format.type === type;
+    });
+    if (!requestedFormat) {
+      return this._getDefaultFormat();
+    }
+    return requestedFormat;
```
Comment (jsanz) [inline src/file_layer.ts:152]: Would this simplify the return?

```suggestion
    return requestedFormat || this._getDefaultFormat;
```

File: src/file_layer.ts [hunk 2]
```diff
@@ -101,7 +119,20 @@ export class FileLayer extends AbstractEmsService {
...
+    const url = this._proxyPath + this._getAbsoluteUrl(format.url);
+    return this._emsClient.extendUrlWithParams(url);
+  }
+
+  private _getFormatMeta(format: EMSFormats) {
+    if ('meta' in format) {
+      return format.meta;
+    } else {
+      return;
+    }
```
Comment (jsanz) [inline src/file_layer.ts:132]: Would it work simply with the optional chaining?

```suggestion
    return format?.meta
```

**Review by jsanz — CHANGES_REQUESTED** [review]
Added a couple of small suggestions in the code.

Apart from that, should the new functions have their own tests coverage or with the changes in the current tests covering the two formats case, and the simple logic behind is enough to not need them?

Maybe the rules for the manifest make it hard to have scenarios with for example a repeated format, or a missing key, and so on.

If that's the case, simply go ahead with the merge but I wanted to be sure we are intentional.

File: src/file_layer.ts (hunk 2 above)
Comment (nickpeihl) [reply src/file_layer.ts:132]: I think optional chaining in your suggestion only tests if `format` is defined. In our case, we know `format` is defined, but we don't know if the `meta` property of `format` is defined.

**Comment by nickpeihl** [comment]
> Apart from that, should the new functions have their own tests coverage or with the changes in the current tests covering the two formats case, and the simple logic behind is enough to not need them?

Good idea. I've modified the tests to check against a single format layer and a multiple format layer.

**Review by jsanz — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**